src/test/java/org/candlepin/pki/certs/SCACertificateGeneratorTest.java

```diff
@@ -163,18 +169,646 @@ private String urlEncode(String text) {
     }
 
     @Test
-    public void shouldCreateCertificate() {
-        Owner owner = createOwner();
-        Consumer consumer = createConsumer(owner);
-        when(this.v3CapabilityCheck.isCertV3Capable(consumer)).thenReturn(true);
+    public void testGetSCAContentPayloadWithNullConsumer() {
+        assertThrows(IllegalArgumentException.class, () ->  {
+            generator.getContentPayload(null);
+        });
+    }
 
-        SCACertificate result = this.generator.generate(consumer);
-        assertNotNull(result);
+    @Test
+    public void testGetSCAContentPayloadWithNullOwner() {
+        Consumer consumer = new Consumer()
+            .setContentAccessMode(ContentAccessMode.ORG_ENVIRONMENT.toDatabaseValue());
+
+        ContentAccessPayload actual = generator.getContentPayload(consumer);
+
+        assertNull(actual);
     }
 
     @Test
-    public void testGeneratedCertificateIncludesOwnerKeyInContainerContentPath() throws Exception {
+    public void testGetSCAContentPayloadWithNonSCAOwner() {
+        Owner owner = this.createNonSCAOwner(TestUtil.randomString("owner-"));
+        Consumer consumer = this.createConsumer(owner);
+
+        ContentAccessPayload actual = generator.getContentPayload(consumer);
+
+        assertNull(actual);
+    }
+
+    @Test
+    public void testGetSCAContentPayloadWithNonV3CertCapableConsumer() {
+        Owner owner = this.createOwner(TestUtil.randomString("owner-"));
+        Consumer consumer = this.createConsumer(owner)
+            .setContentAccessMode(ContentAccessMode.ORG_ENVIRONMENT.toDatabaseValue());
+
+        doReturn(false).when(v3CapabilityCheck)
+            .isCertV3Capable(consumer);
+
+        ContentAccessPayload actual = generator.getContentPayload(consumer);
+
+        assertNull(actual);
+    }
+
+    @Test
+    public void testGetSCAContentPayloadWithDifferentArchAndSupportedArches() throws Exception {
```

Comment:
This is good, but we have three other states I'd like to see explicitly tested: consumer w/no arches specified at all, consumer with only arch, and consumer with only supported arches (parameterized with one, two, etc.).

There are also the cases of what happens when supported arches has duplicates, or overlaps with the base arch fact.

Reply:
Good call! I updated this test to be a parameterized test that takes a matrix of arches and supported arches and verifies that the correct content is added to the SCA content payload.